llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp

```diff
@@ -476,12 +482,38 @@ void SplitGraph::Node::visitAllDependencies(
   }
 }
 
+/// Checks if \p I has MD_callees and if it does, parse it and put the function
+/// in \p Callees.
+///
+/// \returns true if there was metadata and it was parsed correctly. false if
+/// there was no MD or if it contained unknown entries and parsing failed.
+/// If this returns false, \p Callees will contain incomplete information
+/// and must not be used.
+static bool handleCalleesMD(const Instruction &I,
```

Comment:
Still think the call graph analysis should be handling this

Reply:
Can it land as is or do you want me to try changing the CG first?
I'm not sure if that'd need an RFC

Reply:
Can start with it here, but I've always found it to be a flaw. It's also busted for the CGInSCC order with aliases (which I guess we're now less dependent on)